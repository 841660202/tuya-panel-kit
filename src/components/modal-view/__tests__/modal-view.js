import { Text } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ModalView from '../modal-view';

describe('ModalView Component', () => {
  it('basic render', () => {
    const component = renderer.create(<ModalView />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('should update props', () => {
    const component = renderer.create(<ModalView transparent />);
    component.update(<ModalView alignContainer='top' />);
    const { instance } = component.root;
    expect(instance.state.alignContainer).toEqual('top');
  });
  it('should set style', () => {
    const component = renderer.create(<ModalView />);
    const { instance } = component.root;
    instance.setModalStyle('top');
    expect(instance.state.alignContainer).toEqual('top');

    instance.setModalStyle('xxx');
    expect(instance.state.alignContainer).toEqual('xxx');
  });
  it('should set panresponder', () => {
    const component = renderer.create(<ModalView />);
    const { instance } = component.root;
    expect(instance._panResponder.panHandlers.onStartShouldSetResponder()).toBe(true);
    expect(instance._panResponder.panHandlers.onMoveShouldSetResponder()).toBe(false);
  });
  it('should close', () => {
    const component = renderer.create(<ModalView />);
    const { instance } = component.root;
    component.update(<ModalView visible={true} />);
    expect(instance.state.visible).toBe(true);
    instance.closeModalView();
    expect(instance.state.visible).toBe(false);
  });
  it('should custom close', () => {
    const fn = jest.fn();
    const component = renderer.create(<ModalView visible={true} onRequestClose={fn} />);
    const { instance } = component.root;
    instance.closeModalView();
    expect(instance.props.onRequestClose).toHaveBeenCalled();
  });
  it('should update child', () => {
    const fn = jest.fn();
    const component = renderer.create(
      <ModalView visible={true} onRequestClose={fn}>
        <Text>123</Text>
      </ModalView>
    );
    const { instance } = component.root;
    instance.updateChild({ style: { color: '#fff' } });
    expect(instance.state.contentNode).toEqual(<Text style={{ color: '#fff' }}>123</Text>);
  });
});

