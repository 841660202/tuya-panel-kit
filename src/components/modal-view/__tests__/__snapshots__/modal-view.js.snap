// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalView Component basic render 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  onShow={undefined}
  transparent={true}
  visible={false}
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      style={
        Object {
          "position": "absolute",
          "width": 750,
        }
      }
    />
  </View>
</Modal>
`;
